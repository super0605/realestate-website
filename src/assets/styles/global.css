.ant-input:hover {
  border-right-width: 2px !important;
}
