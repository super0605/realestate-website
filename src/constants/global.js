export const tokenKey = "messenger:token";

export const sidebarMenuType = {
  navMenu: "navMenu",
  userMenu: "userMenu",
};

export const subMenuItemUsage = {
  parent: "parent",
  child: "child",
};

export const navTags = {
  NavItem: "NavItem",
  NavDropdown: "NavDropdown",
  ChildNavItem: "ChildNavItem",
};

export const responseCodes = {
  "200": "00",
  "400": "89",
  "403": "98",
  "401": "95",
  "500": "99",
};

export const dummyProperties = [
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
  {
    id: "0",
    title: "Apartment in good condition with terrace and elevator in Sant Pau",
    price: 280000,
    pricePerSqm: 2.178,
    sqm: 101,
    numberOfBathrooms: 3,
    numberOfBedrooms: 2,
    imgUrl:
      "https://dywf8esi8sang.cloudfront.net/pictures/6fda34aba1e02cb82ef9fba94e4475f4_medium.jpg",
  },
];
