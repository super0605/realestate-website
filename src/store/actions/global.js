import { createAction } from "@reduxjs/toolkit";

export const CHANGE_THEME = "[global] change theme";

export const changeThemeAction = createAction(CHANGE_THEME);
