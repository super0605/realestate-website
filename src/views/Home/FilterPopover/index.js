import React, { useState } from "react";
import { Popover } from "antd";

const FilterPopover = ({ children, filterContent: ContentComponent }) => {
  const [visible, setVisible] = useState(false);

  const handleVisibleChange = (val) => {
    setVisible(val);
  };

  function hide() {
    setVisible(false);
  }

  return (
    <Popover
      content={<ContentComponent hide={hide} />}
      placement="bottomLeft"
      trigger="click"
      visible={visible}
      onVisibleChange={handleVisibleChange}
    >
      {children}
    </Popover>
  );
};

export default FilterPopover;
