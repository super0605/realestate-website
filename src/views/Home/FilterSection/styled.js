import styled from "styled-components";
import PBox from "../../../components/PBox";

export const PBoxWrap = styled(PBox)`
  padding: 24px;
  position: fixed;
  z-index: 3;
  top: 66px;
  left: 0px;
  right: 0px;
  width: 100%;
  border-top: 1px solid #d9d9d9;
`;

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const FilterGroup = styled.div`
  display: flex;
`;

export const FilterElmWrap = styled.div`
  padding-left: 24px;
`;

export const LocationInfo = styled.div`
  font-size: 20px;
  font-family: "Roboto";
  font-weight: 600;
`;