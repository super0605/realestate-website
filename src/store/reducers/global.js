import { createReducer } from "@reduxjs/toolkit";
import { CHANGE_THEME } from "../actions/global";

export const initialState = {
  theme: "dark"
};

const global = createReducer(initialState, {
  [CHANGE_THEME]: (state, { payload }) => {
    state.theme = payload;
  }
});

export default global;
