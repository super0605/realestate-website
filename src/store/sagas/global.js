import { takeLatest, put } from "redux-saga/effects";

import { CHANGE_THEME } from "../actions/global";

// TODO: will remove later
function* getGlobalSaga(action) {
  try {
    yield put({
      type: "TEST_REDUX_SAGA",
      payload: action.payload
    });
  } catch (e) {
    console.log("error: ", e);
  }
}

export default function*() {
  yield takeLatest(CHANGE_THEME, getGlobalSaga);
}
