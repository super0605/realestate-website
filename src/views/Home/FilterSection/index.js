import React from "react";
import { withTheme } from "styled-components";
import { Button } from "antd";
import { FilterOutlined } from "@ant-design/icons";
import { RangeFilter, CountFilter } from "../FilterContent";
import FilterPopover from "../FilterPopover";
import { Container, FilterElmWrap, PBoxWrap, FilterGroup, LocationInfo } from "./styled";

const FilterSection = ({ theme }) => {
  return (
    <PBoxWrap>
      <Container>
        <LocationInfo>Barcelona</LocationInfo>
        <FilterGroup>
          <FilterElmWrap>
            <FilterPopover filterContent={RangeFilter}>
              <Button>
                Price <FilterOutlined />
              </Button>
            </FilterPopover>
          </FilterElmWrap>
          <FilterElmWrap>
            <FilterPopover filterContent={RangeFilter}>
              <Button>
                Sqm <FilterOutlined />
              </Button>
            </FilterPopover>
          </FilterElmWrap>
          <FilterElmWrap>
            <FilterPopover filterContent={CountFilter}>
              <Button>
                Number of bedrooms <FilterOutlined />
              </Button>
            </FilterPopover>
          </FilterElmWrap>
          <FilterElmWrap>
            <FilterPopover filterContent={CountFilter}>
              <Button>
                Number of bathrooms <FilterOutlined />
              </Button>
            </FilterPopover>
          </FilterElmWrap>
        </FilterGroup>
      </Container>
    </PBoxWrap>
  );
};

export default withTheme(FilterSection);
