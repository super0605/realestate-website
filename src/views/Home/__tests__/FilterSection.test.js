import * as React from "react";
import { mount, shallow } from "enzyme";
import FilterSection from "../FilterSection";
import FilterPopover from "../FilterPopover";
import { defaultFilterOption } from "../../../constants/global";
import { setHookState } from "../../../utils/TestUtils";

describe("FilterSection", () => {
  let props;
  let mountedFilterSection;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedFilterSection) {
      mountedFilterSection = bShallow
        ? shallow(<FilterSection {...props} />)
        : mount(<FilterSection {...props} />);
    }
    return mountedFilterSection;
  };

  const mockHandleFilterOptions = jest.fn();

  beforeEach(() => {
    props = {
      handleFilterOptions: mockHandleFilterOptions,
    };

    React.useState = setHookState({
      filterOptions: defaultFilterOption,
      showFilterBtn: false,
      showLocationInfo: true,
    });

    wrapper = renderWrapper(true);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render filter group when clicking filter button", () => {
    const filterBtn = wrapper.find(".filter-btn");
    filterBtn.simulate("click");
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 5 filter button after clicking filter button", () => {
    expect(wrapper.find(FilterPopover).length).toBe(5);
  });

  it("should render price filter", () => {
    expect(wrapper.find(".filter-price").length).toBe(1);
  });

  it("should render pricePerSqm filter", () => {
    expect(wrapper.find(".filter-pricePerSqm").length).toBe(1);
  });

  it("should render sqm filter", () => {
    expect(wrapper.find(".filter-sqm").length).toBe(1);
  });

  it("should render number of bedrooms filter", () => {
    expect(wrapper.find(".filter-numberOfBedrooms").length).toBe(1);
  });
  
  it("should render number of bathrooms filter", () => {
    expect(wrapper.find(".filter-numberOfBathrooms").length).toBe(1);
  });
});
