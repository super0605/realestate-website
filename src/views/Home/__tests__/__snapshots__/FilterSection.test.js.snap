// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSection should render filter group when clicking filter button 1`] = `
<Styled(PBox)>
  <styled.div>
    <styled.div>
      Barcelona
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <Button
            block={false}
            ghost={false}
            htmlType="button"
            loading={false}
            onClick={[Function]}
          >
            ALL 
            <ForwardRef(FilterOutlined) />
          </Button>
        </styled.div>
        <styled.div>
          <FilterPopover
            appliedFilter={[Function]}
            endVal={1000000}
            filterContent={[Function]}
            maxVal={1000000}
            minVal={0}
            startVal={0}
            title="Price"
            unit="€"
            valueKey="price"
          >
            <Button
              block={false}
              className="filter-price"
              ghost={false}
              htmlType="button"
              loading={false}
            >
              Price
               
              <ForwardRef(FilterOutlined) />
            </Button>
          </FilterPopover>
        </styled.div>
        <styled.div>
          <FilterPopover
            appliedFilter={[Function]}
            endVal={10000}
            filterContent={[Function]}
            maxVal={10000}
            minVal={0}
            startVal={0}
            title="Price Per Sqm"
            unit="€/m²"
            valueKey="pricePerSqm"
          >
            <Button
              block={false}
              className="filter-pricePerSqm"
              ghost={false}
              htmlType="button"
              loading={false}
            >
              Price Per Sqm
               
              <ForwardRef(FilterOutlined) />
            </Button>
          </FilterPopover>
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.div>
          <FilterPopover
            appliedFilter={[Function]}
            endVal={200}
            filterContent={[Function]}
            maxVal={200}
            minVal={0}
            startVal={0}
            title="Square Meter"
            unit="m²"
            valueKey="sqm"
          >
            <Button
              block={false}
              className="filter-sqm"
              ghost={false}
              htmlType="button"
              loading={false}
            >
              Sqm
               
              <ForwardRef(FilterOutlined) />
            </Button>
          </FilterPopover>
        </styled.div>
        <styled.div>
          <FilterPopover
            appliedFilter={[Function]}
            endVal={10}
            filterContent={[Function]}
            maxVal={10}
            minVal={0}
            startVal={0}
            title="Number of Bedrooms"
            unit="rooms"
            valueKey="numberOfBedrooms"
          >
            <Button
              block={false}
              className="filter-numberOfBedrooms"
              ghost={false}
              htmlType="button"
              loading={false}
            >
              Number of bedrooms
               
              <ForwardRef(FilterOutlined) />
            </Button>
          </FilterPopover>
        </styled.div>
        <styled.div>
          <FilterPopover
            appliedFilter={[Function]}
            endVal={10}
            filterContent={[Function]}
            maxVal={10}
            minVal={0}
            startVal={0}
            title="Number of Bathrooms"
            unit="rooms"
            valueKey="numberOfBathrooms"
          >
            <Button
              block={false}
              className="filter-numberOfBathrooms"
              ghost={false}
              htmlType="button"
              loading={false}
            >
              Number of bathrooms
               
              <ForwardRef(FilterOutlined) />
            </Button>
          </FilterPopover>
        </styled.div>
      </styled.div>
      <styled.div>
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
        >
          <i
            className="fa fa-times-circle"
          />
        </Button>
      </styled.div>
    </styled.div>
  </styled.div>
</Styled(PBox)>
`;

exports[`FilterSection should render without problems 1`] = `
<Styled(PBox)>
  <styled.div>
    <styled.div>
      Barcelona
    </styled.div>
    <styled.div>
      <Button
        block={false}
        className="filter-btn"
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
      >
        Filter 
        <ForwardRef(FilterOutlined) />
      </Button>
    </styled.div>
  </styled.div>
</Styled(PBox)>
`;
